Samples/WindowsML/cpp-cmake/Configuration.cmake

```diff
@@ -0,0 +1,35 @@
+#----------------------------------------------------------------------------------------------------------------------
+#
+#----------------------------------------------------------------------------------------------------------------------
+add_library(Configuration INTERFACE)
+
+target_compile_definitions(Configuration
+    INTERFACE
+        UNICODE
+        _UNICODE
+        NOMINMAX
+        WIN32_LEAN_AND_MEAN
+)
+
+target_compile_features(Configuration
+    INTERFACE
+        cxx_std_23
+)
+
+target_compile_options(Configuration
+    INTERFACE
+        $<$<CXX_COMPILER_ID:MSVC>:/d1nodatetime>
```

Comment:
Out of curiosity, why this vs some "if/endif" like outer pair?

Reply:
Force of habit more than anything. And conciseness. When supporting multiple compilers - or multiple compiler front-ends - I find it easier to understand. And when you have, say, `$<$<CXX_COMPILER_ID:MSVC>:$<$<NOT:$<CONFIG:Debug>>:/INCREMENTAL:NO>>` it's easier to spot the 'pattern', rather than having to accommodate the multiple nested scopes otherwise.